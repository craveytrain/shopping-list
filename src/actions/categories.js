export const TypeKeys = {
  SET: 'SET_CATEGORIES',
  ADD: 'ADD_CATEGORY',
  REMOVE: 'REMOVE_CATEGORY',
  NAME: 'SET_CATEGORY_NAME'
};

export const setCategories = categories => ({
  type: TypeKeys.SET,
  categories
});

export const addCategory = category => ({
  type: TypeKeys.ADD,
  category
});

export const removeCategory = id => ({
  type: TypeKeys.REMOVE,
  id
});

export const nameCategory = (id, name) => ({
  type: TypeKeys.NAME,
  id,
  name
});
