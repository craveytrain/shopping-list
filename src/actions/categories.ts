export interface CategoryProps {
  name: string;
  id?: string;
}

export const SET_CATEGORIES = 'SET_CATEGORIES';
export const setCategories = (categories: CategoryProps[]) => ({
  type: SET_CATEGORIES,
  categories
});

export const ADD_CATEGORY = 'ADD_CATEGORY';
export const addCategory = (category: CategoryProps) => ({
  type: ADD_CATEGORY,
  category
});

export const REMOVE_CATEGORY = 'REMOVE_CATEGORY';
export const removeCategory = (id: string) => ({
  type: REMOVE_CATEGORY,
  id
});

export const SET_CATEGORY_NAME = 'SET_CATEGORY_NAME';
export const setCategoryName = (id: string, name: string) => ({
  type: SET_CATEGORY_NAME,
  id,
  name
});
