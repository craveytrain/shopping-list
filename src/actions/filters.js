export const TypeKeys = {
  VISIBILITY: 'VISIBILITY_FILTER'
};

export const visibilityFilter = visibility => ({
  type: TypeKeys.VISIBILITY,
  visibility
});
