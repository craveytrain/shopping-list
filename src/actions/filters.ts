export type Visability = 'SHOW_ALL' | 'SHOW_CHECKED' | 'SHOW_UNCHECKED';

export interface Filters {
  visibility?: Visability;
}

export enum TypeKeys {
  VISIBILITY = 'VISIBILITY_FILTER'
}

interface VisibilityFilter {
  type: TypeKeys.VISIBILITY;
  visibility: Visability;
}

export type ActionTypes = VisibilityFilter;

export const visibilityFilter = (visibility: Visability): VisibilityFilter => ({
  type: TypeKeys.VISIBILITY,
  visibility
});
