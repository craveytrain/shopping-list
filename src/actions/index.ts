export const SET_STATE = 'SET_STATE';
export const setState = (state: object) => ({
  type: SET_STATE,
  state
});
