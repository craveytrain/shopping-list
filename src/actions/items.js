import generateId from 'lib/generateId';

export const TypeKeys = {
  ADD: 'ADD_ITEM',
  REMOVE: 'REMOVE_ITEM',
  TOGGLE: 'TOGGLE_ITEM',
  NAME: 'NAME_ITEM',
  CATEGORIZE: 'CATEGORIZE_ITEM'
};

export const addItem = item => {
  const id = item.id || generateId(typeof item === 'string' ? item : item.name);

  return {
    type: TypeKeys.ADD,
    item: typeof item === 'string' ? { name: item, id } : { ...item, id }
  };
};

export const removeItem = id => ({
  type: TypeKeys.REMOVE,
  id
});

export const toggleItem = id => ({
  type: TypeKeys.TOGGLE,
  id
});

export const nameItem = (id, name) => ({
  type: TypeKeys.NAME,
  id,
  name
});

export const categorizeItem = (id, categoryId) => ({
  type: TypeKeys.CATEGORIZE,
  id,
  categoryId
});
