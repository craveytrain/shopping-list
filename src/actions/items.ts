export const SET_ITEMS = 'SET_ITEMS';
export const setItems = (items: object[]) => ({
  type: SET_ITEMS,
  items
});

export const ADD_ITEM = 'ADD_ITEM';
export const addItem = (item: object) => ({
  type: ADD_ITEM,
  item
});

export const REMOVE_ITEM = 'REMOVE_ITEM';
export const removeItem = (id: string) => ({
  type: REMOVE_ITEM,
  id
});

export const CHECK_ITEM = 'CHECK_ITEM';
export const checkItem = (id: string) => ({
  type: CHECK_ITEM,
  id
});

export const UNCHECK_ITEM = 'UNCHECK_ITEM';
export const uncheckItem = (id: string) => ({
  type: UNCHECK_ITEM,
  id
});

export const SET_ITEM_NAME = 'SET_ITEM_NAME';
export const setItemName = (id: string, name: string) => ({
  type: SET_ITEM_NAME,
  id,
  name
});

export const SET_ITEM_CATEGORY = 'SET_ITEM_CATEGORY';
export const setItemCategory = (id: string, category: string) => ({
  type: SET_ITEM_CATEGORY,
  id,
  category
});
