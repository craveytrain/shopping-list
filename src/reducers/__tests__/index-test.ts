import reducer from 'reducers';
import { setState } from 'actions';
import * as deepFreeze from 'deep-freeze';

describe('index reducer', () => {
  describe('setState', () => {
    it('should set state', () => {
      const initialState = {};

      deepFreeze(initialState);

      const nextState = reducer(initialState, setState({
        categories: [
          {
            name: 'Meat',
            id: 'meat'
          }
        ],
        items: [
          {
            name: 'Steak',
            id: 'steak',
            checked: false,
            category: 'meat'
          }
        ]
      }
      ));

      expect(nextState).toMatchSnapshot();
    });
  });
});
