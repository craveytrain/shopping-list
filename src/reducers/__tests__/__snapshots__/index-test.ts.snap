// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index reducer setState should set state 1`] = `
Object {
  "categories": Array [
    Object {
      "id": "meat",
      "name": "Meat",
    },
  ],
  "items": Array [
    Object {
      "category": "meat",
      "checked": false,
      "id": "steak",
      "name": "Steak",
    },
  ],
}
`;
